hack/update-deps.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright 2022 The CloudEvents Authors
+# SPDX-License-Identifier: Apache-2.0
+# update-deps.sh - Updates Go dependencies in all directories with go.mod files
+#
+# This script:
+# 1. Finds all directories containing go.mod files
+# 2. Goes into each directory and runs go get -u to update dependencies
+# 3. Runs go mod tidy to clean up the go.mod and go.sum files
+
+set -euo pipefail
+
+# Print script banner
+echo "====================================="
+echo "Go Dependencies Update Script"
+echo "====================================="
+
+# Find all directories with go.mod files
+# TODO (@embano1): currently ignores samples/http due to broken deps
+echo "Finding all directories with go.mod files..."
+DIRS=$(find . -name "go.mod" -not -path "./samples/http/*" -exec dirname {} \;)
+if [ -z "$DIRS" ]; then
+  echo "No go.mod files found!"
+  exit 0
+fi
+
+# Count the number of directories
+DIR_COUNT=$(echo "$DIRS" | wc -l | tr -d ' ')
+echo "Found $DIR_COUNT directories with go.mod files"
+echo
+
+# Process each directory
+COUNTER=1
+for DIR in $DIRS; do
+  echo "[$COUNTER/$DIR_COUNT] Processing $DIR"
+
+  # Change to the directory
+  pushd "$DIR" >/dev/null
+
+  # disable go toolchain switching
```

Comment:
I'm curious, why do we need this?
If we're using the same go version for everything maybe we should add a script to check to make sure all go.mod files use the same version. Or am I missing something that would prevent that?

Reply:
Fixed, sorry I was still working on this :) PR is now ready for review. Couldn't fully test the new script yet because of the "limited" testing capabilities with Github Actions ;) We'll see after merge...